repose-aggregator/core/repose-core-api/src/main/resources/META-INF/schema/system-model/system-model.xsd

```diff
@@ -275,13 +283,153 @@
-                    <html:p>Regex of what uri is allowed access.</html:p>
+                    <html:p>
+                        Deprecated: This attribute is deprecated and will be removed in future major release.
+                        Regex of what uri is allowed access.
+                    </html:p>
                 </xs:documentation>
             </xs:annotation>
         </xs:attribute>
+        <xs:assert vc:minVersion="1.1"
+                   xpathDefaultNamespace="##targetNamespace"
+                   test="not(@uri-regex and uri)"
+                   xerces:message="Cannot define both a deprecated uri-regex attribute and the new uri element"
+                   saxon:message="Cannot define both a deprecated uri-regex attribute and the new uri element"/>
+    </xs:complexType>
+
+
+    <xs:complexType name="FilterCriterion" abstract="true">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>Defines the parent of all the conditional filter processing elements.</html:p>
+            </xs:documentation>
+        </xs:annotation>
+    </xs:complexType>
+    <xs:complexType name="Method">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>Defines the HTTP method conditional processing element.</html:p>
+            </xs:documentation>
+        </xs:annotation>
+        <xs:complexContent>
+            <xs:extension base="mod:FilterCriterion">
+                <xs:attribute name="value" type="xs:string">
+                    <xs:annotation>
+                        <xs:documentation>
+                            <html:p>
+                                The HTTP method that is being tested for.
+                            </html:p>
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+            </xs:extension>
+        </xs:complexContent>
+    </xs:complexType>
+    <xs:complexType name="Header">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>Defines the HTTP header conditional processing element.</html:p>
+            </xs:documentation>
+        </xs:annotation>
+        <xs:complexContent>
+            <xs:extension base="mod:FilterCriterion">
+                <xs:attribute name="name" type="xs:string" use="required">
+                    <xs:annotation>
+                        <xs:documentation>
+                            <html:p>
+                                The case insensitive name of the header being tested for.
+                            </html:p>
+                        </xs:documentation>
+                    </xs:annotation>
+                </xs:attribute>
+                <xs:attribute name="value" type="xs:string">
+                    <xs:annotation>
+                        <xs:documentation>
+                            <html:p>
+                                The case sensitive value of the header being tested for.
```

Comment:
Did we decide whether or not we were going to do regexs?

Reply:
For the headers, I thought we decided that it would be explicit.